### ISSUE TITLE
Loss of sub-second granularity in`docker stats` v1.11.2 onwards

### ISSUE THREAD

**Issue by aaithal** [docker/docker#25078]
**Output of `docker version`:**

```
$ docker version
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:47:50 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
$ docker info
Containers: 2
 Running: 1
 Paused: 0
 Stopped: 1
Images: 53
Server Version: 1.11.2
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 42
 Dirperm1 Supported: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 3.13.0-92-generic
Operating System: Ubuntu 14.04.4 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.67 GiB
Name: ip-172-31-14-187
ID: 3FFQ:UTXU:AUCN:D5X5:G7AB:G4HM:TIIU:NYAF:KZBQ:UJPI:J2SE:MBNR
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS

**Steps to reproduce the issue:**
***Behavior on v1.9.1***
1. Install docker version 1.9.1
2. Run `docker run --name nginx -d nginx:latest`
3. Run the golang program listed at the end of the issue description 
4. Sample output
```
$ go run dockerstats.go
stat timestamp:  2016-07-26 15:54:55.86802873 +0000 UTC
stat timestamp:  2016-07-26 15:54:56.868084679 +0000 UTC
stat timestamp:  2016-07-26 15:54:57.868035428 +0000 UTC
stat timestamp:  2016-07-26 15:54:58.868232129 +0000 UTC
stat timestamp:  2016-07-26 15:54:59.868223925 +0000 UTC
```

***Behavior on v1.11.2***
1. Upgrade to docker v1.11.2
2. Run the golang program listed at the end of the issue description 
3. Sample output
```
$ go run dockerstats.go
stat timestamp:  2016-07-26 15:57:02 +0000 UTC
stat timestamp:  2016-07-26 15:57:03 +0000 UTC
stat timestamp:  2016-07-26 15:57:04 +0000 UTC
stat timestamp:  2016-07-26 15:57:05 +0000 UTC
stat timestamp:  2016-07-26 15:57:06 +0000 UTC
```

**Describe the results you received:**
We are loosing the micro/millisecond granularity on metrics that are being streamed from the docker daemon. This can adversely affect the metrics collection/aggregation for docker clients, which would be using the `Read` field to [compute utilization percentages](https://github.com/aws/amazon-ecs-agent/issues/416#issuecomment-235104129). 

I even built docker binaries from the master from [the commit 8f11896a45aea554935353cb1347cc2aa2529395](https://github.com/docker/docker/commit/8f11896a45aea554935353cb1347cc2aa2529395), which includes the [containerd protobuf integration](https://github.com/docker/docker/commit/29b2714580d085533c29807fa337c2b7a302abb6). But, still got the same behavior as docker v1.11.2.

**Describe the results you expected:**
Timestamp parity across all docker versions. We'd like this behavior to go back to the way it behaved in docker v1.9.1

**Additional information you deem important (e.g. issue happens only occasionally):**
I have also seen that frequency of container stats seems to proportional to the cpu load on the host. The frequency under normal load (<100% load running 1~10 containers) is a metric per second. But, under load (100% CPU, lots of containers(100s)), its much higher: upto 10 per second. Either of these could lead to clients incorrectly aggregating/processing metrics.

***Golang Stats Client***
```go
  package main

  import (
          "fmt"

          docker "github.com/fsouza/go-dockerclient"
  )

  func main() {
          endpoint := "unix:///var/run/docker.sock"
          client, _ := docker.NewClient(endpoint)
          stats := make(chan *docker.Stats)
          cancel := make(chan bool)
          id := "nginx"
          options := docker.StatsOptions{
                  ID:     id,
                  Stats:  stats,
                  Stream: true,
                  Done:   cancel,
          }

          go func() {
                  statsErr := client.Stats(options)
                  if statsErr != nil {
                          fmt.Printf("Error retrieving stats for container %s: %v\n", id, statsErr)
                  }
          }()

          for s := range stats {
                  fmt.Println("stat timestamp: ", s.Read)
          }
  }
```

**Comment by cpuguy83** [comment]
@aaithal can you repro on 1.12? I cannot repro.

**Comment by cpuguy83** [comment]
Closing since I can't reproduce on 1.12 and no response from OP.
Feel free to  comment if you get a chance to check this out.

Thanks!

**Comment by samuelkarp** [comment]
This appears fixed to me with Docker 1.12.  Thanks!
